### PR TITLE
[5.6] JSON assertion bugfixes

### PR DESCRIPTION
**Pull request by Juddling**
I've added tests which illustrate the issue I'm having, and this PR adds a fix.

## Bug Description
JSON Request Response:

```
{"id":200,"data":"Lorem Ipsum..."}
```

Running the following would fail:

```
$response->assertJsonMissing(['id' => 20]);
```

The reason this was failing is because the assertion functions search for: `"id":20` which clearly is present in the response.

If instead we search for:

*  `"id":20,`
*  `"id":20}`
*  `"id":20]`

We guarantee that the value we are searching for is present, and not just a portion of the value.

## Long Term
The existing way that the assertion functions operate, i.e. via string comparison isn't ideal, and I'd suggest refactoring it to work recursively walking through the leaves of the array, and then performing value comparisons. Note that this is likely to be less efficient.

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/TestResponse.php [hunk 1]
```diff
@@ -540,6 +540,27 @@ public function assertJsonMissingExact(array $data)
...
+     */
+    protected function jsonSearchStrings($key, $value)
+    {
+        $needle = substr(json_encode([$key => $value]), 1, -1);
+
+        return [
+            $needle.']',
+            $needle.'}',
+            $needle.',',
+        ];
```
Comment (taylorotwell) [inline src/Illuminate/Foundation/Testing/TestResponse.php]: This won't work for strings?

Comment (Juddling) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: See https://github.com/Juddling/framework/blob/85dbe29d646e790bbdd3f110452e49b07222eec6/tests/Foundation/FoundationTestResponseTest.php#L164 for existing tests which cover searching for strings.

**Comment by taylorotwell** [comment]
I see. Thanks.

**Comment by deleugpn** [comment]
Just opened a PR to revert this as it broke existing feature. See https://github.com/laravel/framework/pull/24900

**Pull request merged**